### ISSUE TITLE
Client does not work with nil properly

### ISSUE THREAD

**Issue by ZyX-I** [neovim/lua-client#24]
When writing Neovim tests try using `helpers.meths.set_var('name', nil)`. This will error out, complaining about wrong number of arguments. I have checked the code and it appears that all helper methods written here pass arguments unchanged (i.e. use `func({some args}, ...)` (`...` means literal `...` here)) and not something like `{...}` (which in case argument is nil will produce `{nil}` which is `{[1]=nil}` which is `{}` because tables cannot hold nil values).

Easiest fix there will be creating special exported table named `nil_tbl` and overriding msgpack dumper to dump this table as nil, this will also fix problem with nil that cannot possibly appear inside tables and not just inside other values.

**Comment by tarruda** [comment]
I'm going to rewrite this module soon(on top of libmpack) and it will export a special NIL table to be used where nil is required. Meanwhile, can you use something like this?

```lua
local msgpack = require('MessagePack')
local NIL = {}
local table_packer = msgpack.packers['table']
msgpack.packers['table'] = function(buffer, tbl)
  if tbl == NIL then return msgpack.packers['nil'](buffer) end
  return table_packer(buffer, tbl)
end
```

**Comment by justinmk** [comment]
`mpack.NIL` was implemented in 5b365e5f8995db9723b3bdc5bfd57c1570704a47
